Tests/Command/ClonePagesCommandTest.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Test\PageBundle\Command;
+
+use Prophecy\Prophecy\ObjectProphecy;
+use Sonata\BlockBundle\Model\BlockManagerInterface;
+use Sonata\PageBundle\Command\ClonePagesCommand;
+use Sonata\PageBundle\Model\PageManagerInterface;
+use Sonata\PageBundle\Model\SiteManagerInterface;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Tester\CommandTester;
+
+/**
+ * Class ClonePagesCommand.
+ *
+ * @author Christian Gripp <mail@core23.de>
+ */
+class ClonePagesCommandTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var Application|ObjectProphecy
+     */
+    private $application;
+
+    /**
+     * @var SiteManagerInterface|ObjectProphecy
+     */
+    private $siteManager;
+
+    /**
+     * @var PageManagerInterface|ObjectProphecy
+     */
+    private $pageManager;
+
+    /**
+     * @var BlockManagerInterface|ObjectProphecy
+     */
+    private $blockManager;
+
+    protected function setUp()
+    {
+        $this->siteManager = $this->prophesize('Sonata\PageBundle\Model\SiteManagerInterface');
+        $this->pageManager = $this->prophesize('Sonata\PageBundle\Model\PageManagerInterface');
+        $this->blockManager = $this->prophesize('Sonata\PageBundle\Model\BlockManagerInterface');
+
+        $container = $this->prophesize('Symfony\Component\DependencyInjection\ContainerInterface');
+        $container->get('sonata.page.manager.site')->willReturn($this->siteManager->reveal());
+        $container->get('sonata.page.manager.page')->willReturn($this->pageManager->reveal());
+        $container->get('sonata.page.manager.block')->willReturn($this->blockManager->reveal());
+
+        $command = new ClonePagesCommand();
+        $command->setContainer($container->reveal());
+
+        $this->application = new Application();
+        $this->application->add($command);
+    }
+
+    public function testExecute()
+    {
+        $sourceSite = $this->prophesize('Sonata\PageBundle\Model\SiteInterface');
+        $destSite = $this->prophesize('Sonata\PageBundle\Model\SiteInterface');
+
+        $this->siteManager->find(23)->willReturn($sourceSite->reveal());
+        $this->siteManager->find(42)->willReturn($destSite->reveal());
+
+        $newPage = $this->prophesize('Sonata\PageBundle\Model\PageInterface');
+
+        $page1 = $this->prophesize('Sonata\PageBundle\Model\PageInterface');
+        $page1->getId()->willReturn(1);
+        $page1->getTitle()->willReturn('Page 1');
+        $page1->getParent()->willReturn(null);
+        $page1->isHybrid()->willReturn(true);
+        $page1->__clone()->willReturn($newPage);
```

Comment:
I does not look like it is possible : https://github.com/phpspec/prophecy/issues/198